src/Credentials.js

```diff
@@ -41,20 +42,45 @@ export default class Credentials {
   // Receive response token from user and return data to promise
   receive (token, callbackUrl = null) {
     return verifyJWT(this.settings, token, callbackUrl).then(({payload, profile}) => {
-      const credentials = {...profile, ...(payload.own || {}), ...(payload.capabilities && payload.capabilities.length === 1 ? {pushToken: payload.capabilities[0]} : {}), address: payload.iss}
-      if (payload.nad) {
-        credentials.networkAddress = payload.nad
-      }
-      if (payload.verified) {
-        return Promise.all(payload.verified.map(token => verifyJWT(this.settings, token))).then(verified => {
-          return {...credentials, verified: verified.map(v => ({...v.payload, jwt: v.jwt}))}
-        })
+      if(this.settings.address) {
+        if(payload.req) {
+          return verifyJWT(this.settings, payload.req).then((challenge) => {
+            if(challenge.payload.iss === this.settings.address) {
+              const credentials = {...profile, ...(payload.own || {}), ...(payload.capabilities && payload.capabilities.length === 1 ? {pushToken: payload.capabilities[0]} : {}), address: payload.iss}
```

Comment:
@aldigjo this code is basically the same as in line 66. I hate code repetition as it's very easy to introduce errors. Try extracting out a simple function for internal use only that you can call these 2 places instead. 

I'm thinking perhaps a local function before line 45 as it would already have profile and payload in it's closure.

e.g..

```js
function processPayload () {
  const credentials = {...profile, ...(payload.own || {}), ...(payload.capabilities && payload.capabilities.length === 1 ? {pushToken: payload.capabilities[0]} : {}), address: payload.iss}
  if (payload.nad) {
    credentials.networkAddress = payload.nad
  }
   if (payload.verified) {
     return Promise.all(payload.verified.map(token => verifyJWT(this.settings, token))).then(verified => {
        return {...credentials, verified: verified.map(v => ({...v.payload, jwt: v.jwt}))}
      })
    } else {
      return credentials
   }
}
```

Does that make sense?